src/pages/dar_application/DataAccessRequest_new.js

```diff
@@ -0,0 +1,330 @@
+import { useState, useEffect, useCallback} from 'react';
+import { a, div, h, h4, span, p } from 'react-hyperscript-helpers';
+import isNil from 'lodash/fp/isNil';
+import { DataSet, DAR } from '../../libs/ajax';
+import { FormField, FormFieldTitle, FormFieldTypes, FormValidators } from '../../components/forms/forms';
+
+const searchOntologies = (query, callback) => {
+  let options = [];
+  DAR.getAutoCompleteOT(query).then(
+    items => {
+      options = items.map(function(item) {
+        return item.label;
+      });
+      callback(options);
+    });
+};
+
+const searchDatasets = (query, callback) => {
+  DataSet.searchDatasets(query).then(items => {
+    let options = items.map(function (item) {
+      return formatSearchDataset(item);
+    });
+    callback(options);
+  });
+};
+
+const formatSearchDataset = (ds) => {
+
+  return {
+    key: ds.dataSetId,
+    value: ds.dataSetId,
+    dataset: ds,
+    displayText: ds.datasetIdentifier,
+    label: span({}, [
+      span({style: {fontWeight: 'bold'}}, [ds.datasetIdentifier]), ' | ', ds.name]),
+  };
+};
+
+const datasetsContainDataUseFlag = (datasets, flag) => {
+  return datasets.some((ds) => ds?.dataUse[flag] === true);
+
+};
+
+const fetchAllDatasets = async (dsIds) => {
+  let datasets;
+  if (!isNil(dsIds)) {
+    datasets = await Promise.all(dsIds.map((id) => DataSet.getDatasetByIdV2(id)));
+  } else {
+    datasets = [];
+  }
+
+  return datasets;
+};
+
+export default function DataAccessRequest(props) {
+  const {
+    formFieldChange,
+    formData,
+    setIrbDocument,
+    setCollaborationLetter,
+    ariaLevel = 2
+  } = props;
+
+  const [datasets, setDatasets] = useState([]);
+  const today = new Date();
+  const irbProtocolExpiration = formData.irbProtocolExpiration || `${today.getFullYear().toString().padStart(4, '0')}-${today.getMonth().toString().padStart(2, '0')}-${today.getDate().toString().padStart(2, '0')}`;
+
+
+  useEffect(() => {
+    fetchAllDatasets(formData.datasetIds).then((datasets) => {
+      setDatasets(datasets);
+    });
+  }, [formData.datasetIds]);
```

Comment:
this is _techincally_ inefficient, since it's refetching the datasets every time the user searches something new in the dataset search box. however, in practice, this seems to be robust/fast enough in my testing.